src/components/list/index.stories.tsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { Meta, StoryObj } from '@storybook/react';
+
+import { getRecipeControls } from '.storybook/preview/controls/recipe';
+
+import { Box } from '../box';
+
+import { List, ListRootProps } from './index';
+
+const Template = ({
+  items,
+  showContainer,
+  longText,
+  ...args
+}: ListRootProps & {
+  items: number;
+  showContainer: boolean;
+  longText: boolean;
+}) => (
+  <List.Root
+    {...args}
+    {...(showContainer
+      ? {
+          border: '1px',
+          borderColor: 'gray.900',
+          bg: 'gray.100',
+          display: 'inline-block',
+        }
+      : {})}
+  >
+    {Array.from({ length: items }, (_, i) => (
+      <List.Item key={i}>
+        Item {i + 1} {longText ? 'lorem ipsum dolor sit amet' : ''}
+      </List.Item>
+    ))}
+  </List.Root>
+);
+
+/**
+ * Use combination of `showContainer` and `longText` to see the difference between `listStylePosition` `inside` and `outside`.
+ * */
+const meta: Meta<typeof Template> = {
+  title: 'Components/Data display/List',
+  render: Template.bind({}),
+
+  decorators: [
+    (Story) => (
+      <Box maxW="4xl">
+        <Story />
+      </Box>
+    ),
+  ],
+
+  args: {
+    size: 'md',
+    items: 5,
+    showContainer: false,
+    longText: false,
+  },
+
+  argTypes: {
+    ...getRecipeControls('list'),
+    as: {
+      control: 'select',
+      options: ['ul', 'ol'],
+    },
+    listStyleType: {
+      control: 'select',
+      options: [
```

Comment:
Not sure if we want to expose all types there was only a subset in the current version to avoid using unintended (by UX) types 
![image](https://github.com/user-attachments/assets/6d846ff8-234a-44cf-bee1-005c6a2c31b5)

Reply:
This was our initiative when we added custom components for `Ordered` and `Unordered` lists. 

We _could_ port this, but it would make `ListProps` into quite a complex type - we'd need to have `listStyleType` dependent on `as`, as well as handling the case where `as` is not present. And, to stay consistent with chakra, we would need to support responsive values and an escape hatch for a CSS variable. I think it's fine to follow CSS here for the sake of simplicity.